book/src/pil/types.md

```diff
@@ -9,13 +9,75 @@ The powdr-pil language has the following types:
 - tuple
 - array
 - function type
-- expression
-- constraint
+- `expr` (expression)
+- `constr` (constraint)
+- `!` ("bottom" or "unreachable" type)
+
+> The `col` type is special in that it is not a proper type, but can be used for declaring symbols.
+> See [Types of Symbols](#types-of-symbols) for details.
+
+Powdr-pil does perform a Hindley-Milner-kind type checking. After type checking, for all symbols
+without an explicit type, their type has to be uniquely determined. If you want a symbol to have
+a generic type, that generic type has to be explicitly specified.
+
+Declarations of generic symbols can constraint type variables to satisfy certain bounds.
```

Comment:
Maybe clarify `generic symbol`/`generic type` a bit? A `generic symbol` refers to a symbol with a generic type right?